src/main/kotlin/org/rust/lang/core/types/infer/TypeInference.kt

```diff
@@ -1482,74 +1482,74 @@ private class RsFnInferenceContext(
         }
     }
 
-    private fun RsPat.extractBindings(type: Ty) {
+    private fun RsPat.extractBindings(type: Ty, ignoreRef: Boolean = false) {
```

Comment:
Maybe better to convert `ignoreRef` to `addRef` (or some better name with similar meaning)?
Looks like it would be closer to rustc desugaring described in RFC [examples](https://github.com/rust-lang/rfcs/blob/master/text/2005-match-ergonomics.md#examples)